### PR TITLE
Exit to WP and WP Experience Step Test

### PR DESCRIPTION
**Pull request by Yashita101**

### CODE REVIEW

File: tests/cypress/integration/newfold-labs/wp-module-onboarding/steps/get-started-experience.cy.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    });
+
+    it('Check if Headers Load', () => {
+        cy.get('.nfd-step-card-heading').should('exist');
+        cy.get('.nfd-step-card-subheading-other').should('exist');
+        cy.get('.nfd-step-card-question').should('exist');
+    });
+
+    it('Check if radio options load', () => {
+        cy.get('.components-radio-control__option').should('exist');
```
Comment (arunshenoy99) [inline tests/cypress/integration/newfold-labs/wp-module-onboarding/steps/get-started-experience.cy.js:21]: This being there does not mean that the `input` is present, please check my tax info PR I've added a better checks for radio buttons there.

File: tests/cypress/integration/newfold-labs/wp-module-onboarding/steps/get-started-experience.cy.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    });
+
+    it('Enable Button based on the radio button selected', () => {
+        cy.get('.nfd-card-button').should('be.disabled');
+        cy.url().should('contain', 'get-started/experience');
+
+        cy.get('[type="radio"]').check('1', { force: true });
+        cy.get('[type=radio]:checked').should('have.css', 'background-color', 'rgb(53, 117, 211)');
+        cy.get('.nfd-card-button').should('not.be.disabled').contains('Continue Setup').click();
+        cy.url().should('contain', 'get-started/site-secondary');
```
Comment (arunshenoy99) [inline tests/cypress/integration/newfold-labs/wp-module-onboarding/steps/get-started-experience.cy.js:38]: We might not have `site-secondary` as the next step in the future, so to handle this the only thing we need to ensure is that on `Continue Setup` the url is not `#/wp-setup/step/get-started/experience`

File: tests/cypress/integration/newfold-labs/wp-module-onboarding/steps/get-started-experience.cy.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+    it('Check if radio options load', () => {
+        cy.get('.components-radio-control__option').should('exist');
+    });
+
+    it('Navigation Buttons Landing on expected pages', () => {
+        cy.get('button').contains('Back').click();
+        cy.url().should('contain', 'get-started/welcome');
+        cy.get('button').contains('Next').click();
+        cy.url().should('contain', 'get-started/experience');
```
Comment (arunshenoy99) [inline tests/cypress/integration/newfold-labs/wp-module-onboarding/steps/get-started-experience.cy.js:28]: We might not have these as the next and back steps in the future, so to handle this the only thing we need to ensure is that on Next and Back the url is not #/wp-setup/step/get-started/experience. This test is also failing for me.

File: tests/cypress/integration/newfold-labs/wp-module-onboarding/steps/get-started-experience.cy.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
+// <reference types="Cypress" />
```
Comment (arunshenoy99) [inline tests/cypress/integration/newfold-labs/wp-module-onboarding/steps/get-started-experience.cy.js:1]: Some more relevant tests can be
1. Testing that the drawer does not open on clicking drawer toggle button.
2. Iterating over all the radio buttons and making sure the correct button is checked on click, Ref: Tax Info PR
3. Ensuring that on initial load the `Continue Setup` button is disabled.

File: tests/cypress/integration/newfold-labs/wp-module-onboarding/steps/get-started-experience.cy.js [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+describe('Start Setup WP Experience Page', function () {
+
+	before(() => {
+		cy.visit('wp-admin/?page=nfd-onboarding&flow=ecommerce#/wp-setup/step/get-started/experience');
+		cy.injectAxe();
```
Comment (arunshenoy99) [inline tests/cypress/integration/newfold-labs/wp-module-onboarding/steps/get-started-experience.cy.js:7]: Ref my prev comment for injecting axe

File: tests/cypress/integration/newfold-labs/wp-module-onboarding/steps/get-started-experience.cy.js [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+	});
+
+    it('Check if Drawer Suppressed', () => {
+        cy.get('.nfd-onboarding-drawer__panel-site-title-container').scrollIntoView().should('not.be.visible');
+    });
+
+    it('Check if Headers Load', () => {
+        cy.get('.nfd-step-card-heading').should('exist');
+        cy.get('.nfd-step-card-subheading-other').should('exist');
+        cy.get('.nfd-step-card-question').should('exist');
```
Comment (arunshenoy99) [inline tests/cypress/integration/newfold-labs/wp-module-onboarding/steps/get-started-experience.cy.js:17]: and be visible

**Review by arunshenoy99 — COMMENTED** [review]
Can we add cy.setCustomerData() and cy.clearCustomerData() before and after all tests else we will get errors like these.
Screenshot 2022-08-26 at 10 49 19 AM

File: tests/cypress/integration/newfold-labs/wp-module-onboarding/steps/exit-to-wordpress.cy.js [hunk 7]
```diff
@@ -0,0 +1,33 @@
+// <reference types="Cypress" />
+
+describe('Exit to WordPress', function () {
+
+    before(() => {
+        cy.visit('wp-admin/?page=nfd-onboarding&flow=ecommerce#/ecommerce/step/products');
```
[the remark this answers is not in the document]
Comment (Yashita101) [reply tests/cypress/integration/newfold-labs/wp-module-onboarding/steps/exit-to-wordpress.cy.js:6]: Apparently iterations are not supported in Cypress, hence I added the immediate last step where the drawer is suppressed

File: tests/cypress/integration/newfold-labs/wp-module-onboarding/steps/get-started-experience.cy.js (hunk 4 above)
Comment (Yashita101) [reply tests/cypress/integration/newfold-labs/wp-module-onboarding/steps/get-started-experience.cy.js:1]: 1 and 3 is covered - Check if the Suppressed Drawer does not open on clicking Toggle Button, Enable Button based on the radio button selected
Have added a few more checks for radio buttons

File: tests/cypress/integration/newfold-labs/wp-module-onboarding/steps/get-started-experience.cy.js (hunk 3 above)
Comment (Yashita101) [reply tests/cypress/integration/newfold-labs/wp-module-onboarding/steps/get-started-experience.cy.js]: Made the modifications accordingly, and the testcases are passing

File: tests/cypress/integration/newfold-labs/wp-module-onboarding/steps/get-started-experience.cy.js (hunk 2 above)
Comment (Yashita101) [reply tests/cypress/integration/newfold-labs/wp-module-onboarding/steps/get-started-experience.cy.js]: agreed, have made the modification, thanks

File: tests/cypress/integration/newfold-labs/wp-module-onboarding/steps/exit-to-wordpress.cy.js (hunk 7 above)
[the remark this answers is not in the document]
Comment (arunshenoy99) [reply tests/cypress/integration/newfold-labs/wp-module-onboarding/steps/exit-to-wordpress.cy.js:7]: I understand in the interest of time, this and a few other reasons are exactly why I mentioned we can handle this in later iterations.

File: tests/cypress/integration/newfold-labs/wp-module-onboarding/steps/exit-to-wordpress.cy.js [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+    it('Check if heading and paragh content exists', () => {
+        cy.get('h1.components-modal__header-heading').should('be.visible');
+        cy.get('.components-modal__content > p').should('be.visible');
+    });
+
+    it("Stay on Onboarding Page when 'X'/Continue is clicked", () => {
+        var currUrl = cy.url().should('not.contain', '/ecommerce/step/products');
+
+        cy.get('.components-modal__header > .components-button').click();
+        currUrl;
```
Comment (arunshenoy99) [inline tests/cypress/integration/newfold-labs/wp-module-onboarding/steps/exit-to-wordpress.cy.js:35]:
```suggestion

```

File: tests/cypress/integration/newfold-labs/wp-module-onboarding/steps/exit-to-wordpress.cy.js [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+        var noExitToWPLabel = cy.get('.nfd-onboarding-drawer__panel-back').should('not.contain', 'Exit');
+        if(drawer_status || noExitToWPLabel) {
+            cy.get('.navigation-buttons_next').click();
+        }
+        cy.url().should('not.contain', '/ecommerce/step/products');
+        cy.get('.nfd-onboarding-etw__trigger').click();
+        cy.get('.components-modal__content').should('be.visible');
+    });
+
+    it('Check if heading and paragh content exists', () => {
```
Comment (arunshenoy99) [inline tests/cypress/integration/newfold-labs/wp-module-onboarding/steps/exit-to-wordpress.cy.js:26]:
```suggestion
    it('Check if heading and paragraph content exists', () => {
```

File: tests/cypress/integration/newfold-labs/wp-module-onboarding/steps/exit-to-wordpress.cy.js [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+        cy.get('h1.components-modal__header-heading').should('be.visible');
+        cy.get('.components-modal__content > p').should('be.visible');
+    });
+
+    it("Stay on Onboarding Page when 'X'/Continue is clicked", () => {
+        var currUrl = cy.url().should('not.contain', '/ecommerce/step/products');
```
Comment (arunshenoy99) [inline tests/cypress/integration/newfold-labs/wp-module-onboarding/steps/exit-to-wordpress.cy.js:32]:
```suggestion
               cy.url().then((currUrl) => {
            cy.get('.components-modal__header > .components-button').click();

            cy.get('.nfd-onboarding-etw__trigger').click();
            cy.get('.nfd-onboarding-etw__buttons > .is-secondary').click();
            cy.url().should('equal', currUrl);
        });
        });
```

Comment (arunshenoy99) [reply tests/cypress/integration/newfold-labs/wp-module-onboarding/steps/exit-to-wordpress.cy.js:32]: This is what I meant in reference to the previous comment (can't find it looks like it was marked as resolved). Please fix the bad indentation as you push the change, thanks.

File: tests/cypress/integration/newfold-labs/wp-module-onboarding/steps/exit-to-wordpress.cy.js [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+
+        cy.get('.components-modal__header > .components-button').click();
+        currUrl;
+
+        cy.get('.nfd-onboarding-etw__trigger').click();
+        cy.get('.nfd-onboarding-etw__buttons > .is-secondary').click();
+        currUrl;
```
Comment (arunshenoy99) [inline tests/cypress/integration/newfold-labs/wp-module-onboarding/steps/exit-to-wordpress.cy.js:39]:
```suggestion

```

File: tests/cypress/integration/newfold-labs/wp-module-onboarding/steps/exit-to-wordpress.cy.js [hunk 12]
```diff
@@ -0,0 +1,33 @@
...
+        cy.get('.is-secondary').contains('Continue').click();
+
+        cy.url().should('contain', '#/wp-setup/step/basic-info');
+    });
+
+    it('Exit to WordPress Page', () => {
+        cy.get('.is-drawer-fade > button').contains('Exit to WordPress').click();
+        cy.get('.is-primary').contains('Exit').click();
+        cy.wait(3000);
+        cy.url().should('contain', '#/home/onboarding');
```
[the remark this answers is not in the document]
Comment (arunshenoy99) [reply tests/cypress/integration/newfold-labs/wp-module-onboarding/steps/exit-to-wordpress.cy.js:46]: Still failing tagging you in a comment on AB's PR, I've figured out why

**Review by arunshenoy99 — APPROVED** [review]
We can loop through the radio buttons and make sure every single one is working similar to the tax info step in future iterations

**Review by arunshenoy99 — CHANGES_REQUESTED** [review]
https://user-images.githubusercontent.com/38878906/187740763-23725e85-d3f2-4d1d-a647-ded22f6c769b.mp4
Ran this on a docker env similar to Github today, attaching the video of failing tests.

**Comment by arunshenoy99** [comment]
Pushed the fix.

**Review by officiallygod — APPROVED** [review]

**Review by arunshenoy99 — APPROVED** [review]
Made the change

**Pull request merged**